numpyro/distributions/transforms.py

```diff
@@ -1200,6 +1201,80 @@ def __eq__(self, other):
         )
 
 
+def _normalize_rfft_shape(input_shape, shape):
+    if shape is None:
+        return input_shape
+    return input_shape[:len(input_shape) - len(shape)] + shape
+
+
+class RealFastFourierTransform(Transform):
+    """
+    N-dimensional discrete fast Fourier transform for real input.
+
+    :param shape: Length of each transformed axis to use from the input, defaults to the
+        input size.
+    :param ndims: Number of trailing dimensions to transform.
+    """
+
+    domain = constraints.dependent
+    codomain = constraints.dependent
+
+    def __init__(
+        self,
+        shape=None,
+        ndims=1,
```

Comment:
maybe using `event_dim` to match patterns in constraints. Could you elaborate on why we need this attribute?

Reply:
The motivation is to specify the number of dimensions along which to transform, e.g., `ndim=1` gives the one-dimensional rfft, `ndim=2` gives the two-dimensional rfft, etc. Maybe `event_ndims` or `transform_ndims` for consistency with `reinterpreted_batch_ndims` in `IndependentTransform`?

Reply:
Oh, I didn't notice that you are supporting n-dimensional fft. Thanks for explaning!
Re event_ndims: looks better to me. We used `event_dim` just to be consistent with initial distribution designs.

Reply:
Actually, I like `transform_shape` and `transform_ndims` better. It is up to you. :)

Reply:
I've kept it with `transform_ndims` and `transform_shape` for now. I was thinking `shape` could be good because that's the terminology numpy uses (although they just call the parameter `s`).